crypto/evp/p_rsa.c

```diff
@@ -569,7 +570,7 @@ static int pkey_rsa_ctrl(EVP_PKEY_CTX *ctx, int type, int p1, void *p2) {
       }
       // Check if the hashAlgorithm is matched.
       // Sec 3.3 https://tools.ietf.org/html/rfc4055#section-3.3
-      if (!pss_hash_algorithm_match(ctx, rctx->md, p2)) {
+      if (!pss_hash_algorithm_match(ctx, rctx->min_saltlen, rctx->md, p2)) {
```

Comment:
Nit: could pass reference to `rctx`.

Reply:
This does not work when checking `mgf1md`

```
if (!pss_hash_algorithm_match(ctx, rctx->min_saltlen, rctx->mgf1md, p2)) {
```